CHANGELOG.md

```diff
@@ -25,6 +25,8 @@ _ `_optional` subpackage for managing optional dependencies
   `DiscreteCardinalityConstraint`/`ContinuousCardinalityConstraint` subclasses
```

Comment:
80% of this PR is deprecation code

I often wonder why you're imposing this on yourself (us) because
- additional writing
- additional reviewing
- additional maintenance
- additional tests
- has to be removed

As a result I don't think it was a wise choice in this instance. Deprecation should be there for important interfaces expected to be used by many users. `Searchspace.transform` is not one of them.

Reply:
Yeah, perhaps we could have skipped it here .... and agree, it was more work than expected. At least it's another reminder to us to be more careful with the interfaces when we write new code.

Now that it's already there, shall we keep it or kick it?

Also, regardless of the deprecation: do you agree with the new interface?

Reply:
have renamed to `transform` and change the signature

Reply:
all good for me